### PR TITLE
Post processed titles

### PR DESCRIPTION
**Pull request by cgsandford**
Adds "Post-Processed" as a prefix to cube title attributes where required.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
- improver/__init__.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
- improver/__init__.py  3
         

Clones removed
==============
+ improver_tests/threshold/test_BasicThreshold.py  -7
+ improver_tests/utilities/test_OccurrenceWithinVicinity.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
- improver/__init__.py  3
         

Clones removed
==============
+ improver_tests/utilities/test_OccurrenceWithinVicinity.py  -2
+ improver_tests/threshold/test_BasicThreshold.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/__init__.py  3
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
         

Clones removed
==============
+ improver_tests/utilities/test_OccurrenceWithinVicinity.py  -2
+ improver_tests/threshold/test_BasicThreshold.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
- improver/__init__.py  3
         

Clones removed
==============
+ improver_tests/utilities/test_OccurrenceWithinVicinity.py  -2
+ improver_tests/threshold/test_BasicThreshold.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

File: improver/__init__.py [hunk 1]
```diff
@@ -53,3 +54,33 @@ def __call__(self, *args, **kwargs):
...
+
+    @staticmethod
+    def post_processed_title(cube):
+        """Updates title attribute on output cube"""
+        default_title = MANDATORY_ATTRIBUTE_DEFAULTS["title"]
+        if ("title" in cube.attributes.keys() and
+                cube.attributes["title"] != default_title and
+                "Post-Processed" not in cube.attributes["title"]):
+            title = cube.attributes["title"]
+            cube.attributes["title"] = "Post-Processed {}".format(title)
```
Comment (MoseleyS) [inline improver/__init__.py]:
```suggestion
            cube.attributes["title"] = f"Post-Processed {title}"
```

File: improver/__init__.py [hunk 2]
```diff
@@ -53,3 +54,33 @@ def __call__(self, *args, **kwargs):
...
+            iris.cube.Cube:
+                Output of self.process() with updated title attribute
+        """
+        cube = super().__call__(*args, **kwargs)
+        self.post_processed_title(cube)
+        return cube
+
+    @staticmethod
+    def post_processed_title(cube):
+        """Updates title attribute on output cube"""
```
Comment (MoseleyS) [inline improver/__init__.py]:
```suggestion
        """Updates title attribute on output cube to include "Post-Processed"."""
```

File: improver_tests/threshold/test_BasicThreshold.py [hunk 3]
```diff
@@ -169,34 +169,49 @@ class Test_process(IrisTest):
...
+        result = self.plugin(self.cube)
+        self.assertEqual(result.attributes["title"], expected_title)
+
+    def test_title_preserved(self):
+        """Test title is preserved already correct"""
+        expected_title = "Post-Processed UKV Model Forecast"
+        self.cube.attributes["title"] = expected_title
+        result = self.plugin(self.cube)
+        self.assertEqual(result.attributes["title"], expected_title)
+
```
Comment (MoseleyS) [inline improver_tests/threshold/test_BasicThreshold.py]: I think this is the wrong place for these two tests. They are testing the functionality of the PostProcessingPlugin class, not the BasicThreshold class. Add a new file `improver_tests/test_PostProcessingPlugin.py` and build them in there, probably with an empty plugin.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/__init__.py  3
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
         

Clones removed
==============
+ improver_tests/threshold/test_BasicThreshold.py  -7
+ improver_tests/utilities/test_OccurrenceWithinVicinity.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

File: improver_tests/threshold/test_BasicThreshold.py (hunk 3 above)
Comment (cgsandford) [reply improver_tests/threshold/test_BasicThreshold.py]: OK.  I don't think it hurts to have these particular tests in place (at least the "test title is updated" one), but I will create a module explicitly to test the `PostProcessingPlugin`.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/__init__.py  3
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
- improver_tests/test_PostProcessingPlugin.py  1
         

Clones removed
==============
+ improver_tests/utilities/test_OccurrenceWithinVicinity.py  -2
+ improver_tests/threshold/test_BasicThreshold.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
- improver/__init__.py  3
- improver_tests/test_PostProcessingPlugin.py  1
         

Clones removed
==============
+ improver_tests/threshold/test_BasicThreshold.py  -7
+ improver_tests/utilities/test_OccurrenceWithinVicinity.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/__init__.py  3
- improver_tests/standardise/test_AdjustLandSeaPoints.py  1
- improver_tests/test_PostProcessingPlugin.py  1
         

Clones removed
==============
+ improver_tests/utilities/test_OccurrenceWithinVicinity.py  -2
+ improver_tests/threshold/test_BasicThreshold.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5161603&bid=16947653)

**Pull request merged**